pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -519,6 +521,12 @@ func (p *Provisioner) reconcileCluster(cluster *v3.Cluster, create bool) (*v3.Cl
 		cluster.Status.CACert = caCert
 		resetRkeConfigFlags(cluster, updateTriggered)
 
+		if cluster.Status.AppliedSpec.RancherKubernetesEngineConfig != nil {
+			cluster.Status.NodeVersion++
```

Comment:
Is it intended to have this update if we error? This loop runs 20 times potentially

Reply:
I think this check should move below the update call, once that succeeds then call `reconcileForUpgrade`

Reply:
To clarify, update the nodeVersion, call update, if that succeeds then call reconcileForUpgrade